### PR TITLE
Scalock

### PR DESCRIPTION
**Pull request by anshupande**

### CODE REVIEW

**Comment by matthewdfuller** [comment]
Are we creating another database for this? Is it possible to use the existing flight director RDS instance?

File: v3/fleet-local/proxy/scalock-gateway@.service [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+Description=Install Scalock Gateway
+After=docker.service bootstrap.service
+Requires=docker.service
+
+
+[Service]
+User=core
+TimeoutStartSec=0
+Environment="dbpassword=etcdctl get /scalock/dbpassword"
+Environment="dbusername=etcdctl get /scalock/dbpassword"
```
Comment (matthewdfuller) [inline v3/fleet-local/proxy/scalock-gateway@.service]: This should be `/scalock/dbusername`

**Comment by matthewdfuller** [comment]
Can you move all `/scalock/xyz` into `/scalock/config/xyz`?

File: v3/fleet-local/proxy/scalock-gateway@.service (hunk 1 above)
Comment (matthewdfuller) [inline v3/fleet-local/proxy/scalock-gateway@.service]: Let's also change these to `db-password` and `db-username`

File: v3/fleet-local/proxy/scalock-gateway@.service [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Environment="scalockdbname=etcdctl get /scalock/dbname"
+Environment="scalockdb_endpoint=etcdctl get /scalock/scalockdb_endpoint"
```
Comment (matthewdfuller) [inline v3/fleet-local/proxy/scalock-gateway@.service]: Change to: `/scalock/config/db-endpoint`

File: v3/fleet-local/proxy/scalock-gateway@.service [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+ExecStart=/usr/bin/sh -c "source /etc/profile.d/etcdctl.sh && \
+  sudo docker run -d -p 3622:3622 --name scalock-gateway  \
+  -e SCALOCK_DBURL="postgres://$($dbusername):$($dbpassword)@$($scalockdb_endpoint)?sslmode=disable"   \
+  -e SCALOCK_DBUSER=$($dbusername)  \
+  -e SCALOCK_DBPASSWORD=$($dbpassword)   \
+  -e SCALOCK_DBNAME=$($scalockdbname)    \
+  -e SCALOCK_DBHOST=$($scalockdb_endpoint)
+  scalock/scalock-gateway:stable"
```
Comment (matthewdfuller) [inline v3/fleet-local/proxy/scalock-gateway@.service:25]: Change these values as needed (db-xyz)

**Comment by matthewdfuller** [comment]
The equivalent `infrastructure/v2/config/example-stack{all | required}` changes will be:

To "secrets" section:

```
"SCALOCK_PASSWORD": {
    "type": "etcd",
    "path": "/scalock/config/db-password",
    "contents": ""
}
```

To "configs" section:

```
"/scalock/config/db-name": "abc",
"/scalock/config/db-endpoint": "abc",
"/scalock/config/db-username": ""
```

**Comment by anshupande** [comment]
Postgres is the only supported database at this time.
They are looking into the separation of the audit and configuration into two Postgres database instances for either the upcoming version, or the one after that.

**Comment by matthewdfuller** [comment]
Let's hold off on merging this in until the infrastructure repository is updated to include an actual RDS instance

**Pull request closed without merging**